ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-instance-select-answer/activity-instance-select-answer.component.spec.ts

```diff
@@ -0,0 +1,224 @@
+      declarations: [ActivityInstanceSelectAnswer, QuestionPromptComponent],
+      imports: [NoopAnimationsModule, MatFormFieldModule, MatSelectModule, BrowserAnimationsModule],
+      providers: [
+        {
+          provide: ActivityInstanceSelectAnswerService,
+          useValue: {
+            getOptions: () =>
+              of([
+                {
+                  guid: 'ANSWERGUID1',
+                  name: 'Bob',
+                },
+                {
+                  guid: 'ANSWERGUID2',
+                  name: 'Alice',
+                },
+              ]),
+          },
+        },
+      ],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ActivityInstanceSelectAnswer);
+
+    expect(fixture instanceof ComponentFixture).toBeTrue();
+    expect(fixture.componentInstance instanceof ActivityInstanceSelectAnswer).toBeTrue();
+
+    block = new ActivityInstanceSelectQuestionBlock();
+    block.stableId = 'STABLE_ID';
+
+    component = fixture.componentInstance;
+    component.readonly = false;
+    component.block = block;
+  });
+
+  it('should create without errors', () => {
+    expect(component).toBeDefined();
+  });
+
+  it('renders question prompt if one is provided', () => {
+    const questionPrompt = 'How are you?';
+
+    block.question = questionPrompt;
+
+    fixture.detectChanges();
+
+    expect(fixture.debugElement.nativeElement.textContent).toContain(questionPrompt);
+  });
+
+  it('renders label if one is provided', () => {
+    const label = 'Provide your answer';
+
+    block.label = label;
+
+    fixture.detectChanges();
+
+    const el = fixture.debugElement.query(By.css('mat-label'));
+
+    expect(el).toBeDefined();
+    expect(el.nativeElement.textContent).toContain(label);
+  });
+
+  it('renders placeholder if one is provided', () => {
+    const placeholderContent = 'Select option';
+
+    block.placeholder = placeholderContent;
+
+    fixture.detectChanges();
+
+    const matPlaceholder = fixture.debugElement.query(By.css('.mat-select-placeholder'));
+
+    expect(matPlaceholder).not.toBeNull();
+    expect(matPlaceholder.nativeElement.textContent).toContain(placeholderContent);
+  });
+
+  it('disables input if readonly flag is true', () => {
+    component.readonly = true;
+
+    fixture.detectChanges();
+
+    const el = fixture.debugElement.query(By.css('.mat-select'));
+
+    expect(el).not.toBeNull();
+    expect(el.attributes['aria-disabled']).toEqual('true');
+  });
+
+  it('tries to get options on first render', done => {
+    component.options$.subscribe(options => {
+      if (options.length) {
+        done();
+      }
+    });
+
+    component.ngOnInit();
+  });
+
+  it('tries to get options when panel is opened', () => {
+    component.ngOnInit();
+
+    const initialOptions = component.options$.getValue();
+
+    expect(initialOptions.length).toEqual(2);
+
+    (component as any).activityInstanceSelectAnswerService = {
```

Comment:
Interesting way to mock a service 😀

Reply:
I just wanted to update what values are returned from `getOptions` method and couldn't find a better way 😅

Reply:
It is not an issue, just fyi:
we usually use ` jasmine.SpyObj<T>`, ie. like this https://javascript.tutorialink.com/jasmine-createspyobj-with-properties/